core/trino-spi/src/main/java/io/trino/spi/connector/ColumnSchema.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.connector;
+
+import io.trino.spi.type.Type;
+
+import java.util.Objects;
+
+import static io.trino.spi.connector.SchemaUtil.checkNotEmpty;
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+
+public class ColumnSchema
```

Comment:
We may need a build for this, like we have for `ColumnMetadata`

as we strive for this class to be as small as possible, i don't believe we won't extend it in the future with additional information

for example we have nullable, we could add some other contraint information